### PR TITLE
chore(common): build-utils.sh minor tweaks

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixing up a number of minor bits and pieces identified during demo:

* invalid action or :target would not be picked up if paired with a valid one (e.g.: build:foo should not have worked)
* colors defined in both shellHelperFunctions and build-utils were problematic; able to remove them from shellHelperFunctions because all references included both scripts
* builder_has_action and builder_report now allow both `action:target` and `action :target`
* keyman-version/build.sh did not have working test action...

@keymanapp-test-bot skip

### CODE REVIEW

File: resources/build/build-utils.sh [hunk 1]
```diff
@@ -270,21 +270,31 @@ replaceVersionStrings_Mkver() {
 # Standard build script functions for managing command line, actions and targets
 ################################################################################
 
-# TODO: colors are defined here and in shellHelperFunctions.sh
 # The following allows coloring of warning and error lines, but only works if there's a
 # terminal attached, so not on the build machine.
-if [[ -n "$TERM" ]] && [[ "$TERM" != "dumb" ]] && [[ "$TERM" != "unknown" ]]; then
+builder_use_color() {
```
Comment (ermshiperete) [inline resources/build/build-utils.sh:275]: I don't think this method is intended to be used outside of this file, is it?

```suggestion
_builder_use_color() {
```

File: resources/build/build-utils.sh [hunk 2]
```diff
@@ -270,21 +270,31 @@ replaceVersionStrings_Mkver() {
...
     COLOR_YELLOW=
     COLOR_RESET=
+    HEADING_SETMARK=
+  fi
+}
+
+if [[ -n "$TERM" ]] && [[ "$TERM" != "dumb" ]] && [[ "$TERM" != "unknown" ]]; then
+  builder_use_color true
+else
+  builder_use_color false
```
Comment (ermshiperete) [inline resources/build/build-utils.sh:297]:
```suggestion
  _builder_use_color true
else
  _builder_use_color false
```

**Review by ermshiperete — APPROVED** [review]

**Review by ermshiperete — APPROVED** [review]

File: resources/shellHelperFunctions.sh [hunk 3]
```diff
@@ -82,32 +82,9 @@ get_platform_folder() {
...
-}
-
-if [[ -n "$TERM" ]] && [[ "$TERM" != "dumb" ]] && [[ "$TERM" != "unknown" ]]; then
-  shell_helper_color true
-else
-  shell_helper_color false
-fi
+# Note: COLOR_ and HEADING_ variables are defined in build-utils.sh
+# which is already required for scripts so these should be safe.
+# (part of TODO: consolidate these two scripts)
```
[the remark this answers is not in the document]
Comment (ermshiperete) [reply resources/shellHelperFunctions.sh:87]: yes, with the consolidate source package it should now be possible to use the standard include script.

**Review by jahorton — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 16.0.43-alpha](https://keyman.com/downloads/releases/alpha/16.0.43)

**Pull request merged**